src/class.h

```diff
@@ -12,14 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../imageio/image_dec.h"
+#include "../imageio/imageio_util.h"
 #include "webp/encode.h"
+#include "webp/mux.h"
 
 namespace libwebp {
 
-// Class that does something.
+// Takes time stamped images as an input and produces an animation.
 class Thumbnailer {
-  // TODO: add an API to add frames with a timestamp.
-  // TODO: add an API to generate the animation.
+  WebPAnimEncoder* enc = NULL;
+  WebPPicture pic;
+  WebPData webp_data;
+  WebPAnimEncoderOptions anim_config;
+  WebPConfig config;
+  int timestamp = 0;
+  int width;
+  int height;
+  int duration = 500;
+  int loop_count;
+  int num_pic = 0;
+
+  // Initialize
+  Thumbnailer();
+
+  // Add frame with a timestamp
+  int AddFrame(const char* filename);
+
+  // Generate the animation
+  int GenerateAnimation(const char* output);
```

Comment:
what is the returned int ? If it is a status code (please make it an enum), what can happen ?